ql/cashflows/iborcoupon.cpp

```diff
@@ -84,8 +84,30 @@ namespace QuantLib {
         return spanningTimeIndexMaturity_;
     }
 
-    Rate IborCoupon::indexFixing() const {
+    bool IborCoupon::hasFixed(Date refDate) const {
+        if (refDate == Date()) {
+            refDate = QuantLib::Settings::instance().evaluationDate();
+        }
+
+        if (fixingDate_ > refDate) {
+            return false;
+        } else if (fixingDate_ < refDate) {
+            return true;
+        } else {
+            // fixingDate_ == refDate
+            if (QuantLib::Settings::instance().enforcesTodaysHistoricFixings()) {
+                return true;
```

Comment:
This no longer holds once you add the `refDate` argument to the method call.  `enforcesTodaysHistoricFixings` only comes into play when `refDate` is the evaluation date.